aiida/orm/nodes/node.py

```diff
@@ -47,9 +49,18 @@ def __init__(self, fileobj, name):
     def _warn_if_not_entered(self, method):
         """Fire a warning if the object wrapper has not yet been entered."""
         if not self._was_entered:
-            msg = '`{}` used without context manager for {}. This will raise starting from `aiida-core==2.0.0`'.format(
-                method, self._name
-            )
+            msg = f'\n`{method}` called outside Node.open() context manager for {self._name}.\n' + \
+                  'Please wrap this call inside `with <node instance>.open(): ...` to silence this warning. ' + \
+                  'This will raise starting from `aiida-core==2.0.0`\n'
+            try:
+                caller_frame = inspect.currentframe().f_back.f_back
+            except AttributeError:
+                warnings.warn(msg, AiidaDeprecationWarning)
+                return
+
+            caller_traceback = traceback.extract_stack(caller_frame, limit=1)
+            for item in traceback.StackSummary.from_list(caller_traceback).format():
+                msg += item
             warnings.warn(msg, AiidaDeprecationWarning)  # pylint: disable=no-member
```

Comment:
I think putting the one call in try/except helps but is potentially not enough. We really want to avoid exceptions occurring in this handler. I also think the code to extract the relevant line of code can be much simpler. I propose the following for the entire method, which has updated warning message and stack extraction:
```
        if not self._was_entered:
            msg = f'\nThe method `{method}` was called on the return value of `{self._name}.open()` outside of a context manager.\n' + \
                  'Please wrap this call inside `with <node instance>.open(): ...` to silence this warning. ' + \
                  'This will raise an exception, starting from `aiida-core==2.0.0`.\n'

            try:
                caller = traceback.format_stack()[-3]
            except Exception:  # pylint: disable=broad-except
                msg += 'Could not determine the line of code responsible for triggering this warning.'
            else:
                msg += f'The offending call comes from:\n{caller}'

            warnings.warn(msg, AiidaDeprecationWarning)  # pylint: disable=no-member
```

Reply:
sounds good to me!